torchdynamo/variables/tensor.py

```diff
@@ -145,32 +148,26 @@ def context():
                             lambda: cls.run_proxy(proxy, args, kwargs, nnmodule)
                         )
                 except RuntimeError as e:
-                    raise TorchRuntimeError() from e
+                    if use_fake_tensors and isinstance(e, DataDependentOutputException):
+                        if (
+                            config.capture_scalar_outputs
+                            and proxy.node.target == "item"
+                        ):
+                            example_value = torch.zeros(
+                                size=(), dtype=args[0].dtype
```

Comment:
the args here could be malformed, can you assert that:

args len is 1
args[0] is a tensor
args[0] is a numel() of 1

Reply:
This is post-dispatch, so it should already have thrown in that case

Reply:
Ah yea sg